### PR TITLE
lib: implement `setPrio`

### PR DESCRIPTION
**Pull request by oxij**
For when `hiPrio` and `lowPrio` are not enough.

### CODE REVIEW

File: lib/meta.nix [hunk 1]
```diff
@@ -41,16 +41,18 @@ rec {
...
 
 
-  /* Apply a function to each derivation and only to derivations in an attrset
+  /* Apply a function to each derivation and only to derivations in an attrset.
   */
   mapDerivationAttrset = f: set: lib.mapAttrs (name: pkg: if lib.isDerivation pkg then (f pkg) else pkg) set;
 
+  /* Set the nix-env priority of the package.
+  */
+  setPrio = priority: drv: addMetaAttrs { inherit priority; } drv;
```
Comment (Mic92) [inline lib/meta.nix]: is this not basically the same as?

```
addMetaAttrs { priority = 10; } drv;
```

Comment (7c6f434c) [reply lib/meta.nix]: I think this is a no-op readability change; do you think exposing implementation is better?

(for example, if there are other frequent `meta` overrides, improving visibility of `addMetaAttrs` could be indeed useful)

**Comment by oxij** [comment]
> I think this is a no-op readability change; do you think exposing implementation is better?

Well, to me it just seems strange to expose just two particular priority values out of the whole range.

I do use `setPrio` in my config to influence the priority of the manual pages.

> (for example, if there are other frequent `meta` overrides, improving visibility of `addMetaAttrs` could be indeed useful)

I don't think I use any others, though.

**Review by 7c6f434c — APPROVED** [review]

**Comment by oxij** [comment]
ping?

**Pull request merged**